cirq-core/cirq/transformers/transformer_api.py

```diff
@@ -291,22 +309,36 @@ def method_with_logging(
         func = cls_or_func
 
         @functools.wraps(func)
-        def func_with_logging(
-            circuit: 'cirq.AbstractCircuit', context: TransformerContext
-        ) -> 'cirq.AbstractCircuit':
-            return _transform_and_log(func, func.__name__, circuit, context)
+        def func_with_logging(circuit, **kwargs) -> 'cirq.AbstractCircuit':
+            return _transform_and_log(
+                func,
+                func.__name__,
+                circuit,
+                _get_context(func, f'{func.__name__}()', **kwargs),
+                **kwargs,
+            )
 
         return func_with_logging
 
 
+def _get_context(func, name, **kwargs):
+    if 'context' in kwargs:
+        return kwargs['context']
+    sig = inspect.signature(func)
+    default_context = sig.parameters["context"].default
+    assert (
+        default_context != inspect.Parameter.empty
+    ), "`context` argument must have a default value specified."
+    return default_context
+
+
 def _transform_and_log(
-    func: TRANSFORMER,
-    transformer_name: str,
-    circuit: 'cirq.AbstractCircuit',
-    context: TransformerContext,
+    func, transformer_name, circuit, extracted_context, **kwargs
```

Comment:
nit: might be nice to add back the annotations as they were before, with `extracted_context: Optional[TransformerContext]`.

Reply:
added type annotations back here and elsewhere.